platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -17,68 +19,366 @@ components within the platform.
 
 ## Purpose and Context
 
-The introduction of the Threshold Signature Scheme (TSS) requires a new mechanism for managing node participation in
-consensus and block signing. The Roster, a subset of the address book, will provide the data structure and API for this
-mechanism. This proposal provides a specification for the behavior of rosters starting from their creation from the
-Address Book to their terminal state. A roster reaches a terminal state when it is either adopted by the platform or
-replaced by a new roster. The candidate roster will always get adopted on the next software upgrade.
+The term `AddressBook` has been ambiguously used to refer to both the Hedera Address Book and the Platform Address
+Book. The Platform Address Book is a proper subset of the Hedera Address Book that is only concerned with the
+consensus nodes, their consensus weight, gossip endpoints, and related cryptography. The Hedera Address Book
+contains additional information that is not relevant to the platform portion of the consensus nodes, such as the
+account numbers of the nodes, their staked hbar, the proxy endpoints, and the existence of other types of nodes in
+the network. We've decided to disambiguate the two by referring to the Platform Address Book as the `Roster` and
+making their lifecycle distinct from each other. The life-cycle for updating the Roster must also change as a
+pre-requisite to the implementation of Threshold Signature Scheme (TSS) and the generation of network Ledger Ids.
+
+Prior to this proposal, after the completion of the Dynamic Address Book (DAB) Phase 2 effort, the platform
+`AddressBook` is updated by writing out a new config.txt file and updated cryptography to disk in response to a
+freeze upgrade transaction, just before shut down. Out of band, the new config.txt and cryptography are moved to
+replace the previous ones on disk. On software startup, the new config.txt and cryptography are read in and adopted
+during a software upgrade. Discrepancy in the information present between the `Roster` data structure and the
+`AddressBook` data structure make config.txt a poor medium for carrying out `Roster` updates. Writing the updated
+roster out to disk is less secure than storing it in the state. When TSS is implemented, there will be TSS related
+key material stored in the state, and it is imperative that this key material stay in sync with the `Roster` data.
+
+After this proposal is implemented, new rosters will be created and stored in the state for adoption at a software
+upgrade boundary. It has been common practice in testing to use manually created config.txt files to start new
+networks from genesis and transplant merkle state to a different new network from an existing network. To continue to
+allow these processes, alternate mechanisms of providing a genesis roster and a network overriding roster will be
+introduced.
```

Comment:
I think it would be helpful to state explicitly that `config.txt` will no longer be used at all, and that in normal production operation, the `genesis.json` file will be used once at genesis and then no on-disk non-state representations of the address book or roster will be used again.

This is a non-material update, should you agree and decide to add this.

Reply:
`genesis.json` is a new file name and I assume format to me.    This proposal does not suggest switching to a JSON format and this is why we're leaving it with the `X-config.txt` name since we're using the format in `config.txt`.

Reply:
updated wording to reflect your suggestion